src/stream/src/executor/backfill/no_shuffle_backfill.rs

```diff
@@ -277,13 +278,12 @@ where
 
                     // Prefer to select upstream, so we can stop snapshot stream as soon as the
                     // barrier comes.
-                    let backfill_stream =
+                    let mut backfill_stream =
                         select_with_strategy(left_upstream, right_snapshot, |_: &mut ()| {
                             stream::PollNext::Left
                         });
 
-                    #[for_await]
-                    for either in backfill_stream {
+                    while let Some(either) = backfill_stream.next().await {
```

Comment:
Not sure whether they are equivalent?

Reply:
No, the previous one will consume the backfill stream. We want to continue the snapshot read side of the stream after receive barrier, for that, we must still have ownership of the backfill stream.

The new code will ensure we still retain ownership, i.e. backfill_stream is not moved into the `for_await`.

Reply:
The semantics of `for_await` is to consume the iterator by default.

Reply:
You may write `for either in &mut backfill_stream` to avoid consuming it.

Reply:
>  You may write for either in &mut backfill_stream to avoid consuming it. Just like an iterator.


Works, thank you.